src/bidiMapper/domains/context/BrowsingContextProcessor.ts

```diff
@@ -332,59 +332,99 @@ export class BrowsingContextProcessor {
-      // debugger  and ignore them.
-      targetCdpClient
-        .sendCommand('Runtime.runIfWaitingForDebugger')
-        .then(() =>
-          parentSessionCdpClient.sendCommand('Target.detachFromTarget', params)
-        )
-        .catch((error) => this.#logger?.(LogType.debugError, error));
-      return;
-    }
-
     this.#logger?.(
       LogType.debugInfo,
       'AttachedToTarget event received:',
       params
     );
 
-    this.#setEventListeners(targetCdpClient);
-
-    const maybeContext = this.#browsingContextStorage.findContext(
-      targetInfo.targetId
-    );
+    switch (targetInfo.type) {
+      case 'page':
+      case 'iframe': {
+        if (targetInfo.targetId === this.#selfTargetId) {
+          break;
+        }
+
+        this.#setEventListeners(targetCdpClient);
+
+        const cdpTarget = CdpTarget.create(
+          targetInfo.targetId,
+          targetCdpClient,
+          this.#browserCdpClient,
+          sessionId,
+          this.#realmStorage,
+          this.#eventManager,
+          this.#preloadScriptStorage,
+          this.#networkStorage
+        );
 
-    const cdpTarget = CdpTarget.create(
-      targetInfo.targetId,
-      targetCdpClient,
-      this.#browserCdpClient,
-      sessionId,
-      this.#realmStorage,
-      this.#eventManager,
-      this.#preloadScriptStorage,
-      this.#networkStorage
-    );
+        const maybeContext = this.#browsingContextStorage.findContext(
+          targetInfo.targetId
+        );
+        if (maybeContext) {
+          // OOPiF.
+          maybeContext.updateCdpTarget(cdpTarget);
+        } else {
+          // New context.
+          BrowsingContextImpl.create(
+            cdpTarget,
+            this.#realmStorage,
+            targetInfo.targetId,
+            null,
+            this.#eventManager,
+            this.#browsingContextStorage,
+            this.#logger
+          );
+        }
+        return;
+      }
+      case 'worker': {
+        this.#setEventListeners(targetCdpClient);
+
+        const cdpTarget = CdpTarget.create(
+          targetInfo.targetId,
+          targetCdpClient,
+          this.#browserCdpClient,
+          sessionId,
+          this.#realmStorage,
+          this.#eventManager,
+          this.#preloadScriptStorage,
+          this.#networkStorage
+        );
 
-    if (maybeContext) {
-      // OOPiF.
-      maybeContext.updateCdpTarget(cdpTarget);
-    } else {
-      // New context.
-      BrowsingContextImpl.create(
-        cdpTarget,
-        this.#realmStorage,
-        targetInfo.targetId,
-        null,
-        this.#eventManager,
-        this.#browsingContextStorage,
-        this.#logger
-      );
+        const maybeContext = this.#browsingContextStorage.findContext(
+          targetInfo.targetId
+        );
+        if (maybeContext) {
+          // OOPiF.
+          maybeContext.updateCdpTarget(cdpTarget);
+        } else {
+          // New context.
+          BrowsingContextImpl.create(
```

Comment:
this actually looks wrong too since the worker should not have a browsing context